core/client/src/main/java/alluxio/client/file/options/OutStreamOptions.java

```diff
@@ -57,6 +61,13 @@ private OutStreamOptions() {
       throw Throwables.propagate(e);
     }
     mWriteType = conf.getEnum(Constants.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class);
+    try {
+      // Set user and group from user login module, apply default file UMask.
+      mPermissionStatus = PermissionStatus.defaults().setUserFromLoginModule(conf)
+          .applyFileUMask(conf);
+    } catch (IOException e) {
+      throw Throwables.propagate(e);
```

Comment:
This case is a bit different, I'm logging a warning message instead of changing the caller site too much.